LEAF_Nexus/sources/System.php

```diff
@@ -242,4 +256,90 @@ public function refreshOrgchartEmployees()
 
         return $ret;
     }
+
+    /**
+     * Get primary admin.
+     *
+     * @return array|false array with primary admin's info
+     */
+    public function getPrimaryAdmin()
+    {
+        $strSQL = "SELECT `data` FROM `settings` WHERE `setting` = 'primaryAdmin'";
+        $primaryAdminRes = $this->db->query($strSQL);
+
+        include_once 'Employee.php';
+        $employee = new Orgchart\Employee($this->db, $this->login);
+
+        if(count($primaryAdminRes))
+        {
+            $user = $employee->lookupLogin($primaryAdminRes[0]['data']);
+            $user = $employee->lookupEmpUID($user[0]['empUID']);
+            return isset($user[0]) ? $user[0] : $primaryAdminRes[0]['data'];
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Set primary admin.
+     *
+     * @return array array with response array
+     */
+    public function setPrimaryAdmin()
+    {
+        $userID = XSSHelpers::xscrub($_POST['userID']);
+
+        //check if user is system admin
+        include_once 'Employee.php';
+        $employee = new Orgchart\Employee($this->db, $this->login);
+        $user = $employee->lookupLogin($userID);
+
+        $sqlVars = array(':empUID' => $user[0]['empUID']);
+        $strSQL = "SELECT `empUID` FROM `relation_group_employee`".
+                    "WHERE `groupID` = 1 AND empUID = :empUID";
+        $res = $this->db->prepared_query($strSQL, $sqlVars);
+
+        if(count($res))
+        {
+            $sqlVars = array(':userID' => $user[0]['userName']);
+            $strSQL = "INSERT INTO `settings` (`setting`, `data`) VALUES ('primaryAdmin', :userID) ".
+                        "ON DUPLICATE KEY UPDATE `data` = :userID";
+            $this->db->prepared_query($strSQL, $sqlVars);
+
+            $resultArray = array('success' => true, 'response' => $res);
+
+            $primary = $this->getPrimaryAdmin();
+
+            $this->dataActionLogger->logAction(\DataActions::ADD, \LoggableTypes::PRIMARY_ADMIN, [
+                new \LogItem("settings", "setting", 'primaryAdmin'),
+                new \LogItem("settings", "data", $primary["empUID"], $primary["firstName"].' '.$primary["lastName"]),
+            ]);
+        }
+        else
+        {
+            $resultArray = array('success' => false, 'response' => $res);
```

Comment:
I know I harp on "else" statements.  But in this case, any variable you are going to return, you want defined in your "base" code block.  Sooo... I would put this right after you run the select query.  Then just drop the else statement.  Inside the main "if", you assign the different values to the different elements of the array.

Reply:
Removed the else and defined the result after the query